example/src/main/java/org/wordpress/android/fluxc/example/ui/products/WooUpdateProductFragment.kt

```
@@ -373,13 +407,16 @@ class WooUpdateProductFragment : Fragment() {
             this.selectedCategories = data?.getParcelableArrayListExtra(ARG_SELECTED_PRODUCT_CATEGORIES)
         } else if (requestCode == PRODUCT_TAGS_REQUEST_CODE) {
             this.selectedTags = data?.getParcelableArrayListExtra(ARG_SELECTED_PRODUCT_TAGS)
+        } else if (requestCode == PRODUCT_DOWNLOADS_REQUEST_CODE) {
+            this.selectedProductDownloads = data!!.getParcelableArrayListExtra(ARG_PRODUCT_DOWNLOADS)
```

np: we could follow the convention here and make `data` as optional? Something like:
`data?.getParcelableArrayListExtra(ARG_PRODUCT_DOWNLOADS)`

---

if we make it optional, and if for any case it's null, the code will fail silently (the update to the downloadable files will be ignored), IMO it's better to have a crash here (especially that the app is just an example), to be able to identify the cause.
PS: I have to add a check on the resultCode in the condition, to make sure that the action was not cancelled.

---

Sounds good  👍